src/index.ts

```diff
@@ -222,6 +222,20 @@ const formats: IFormat[] = [
     },
     name: 'BNB',
   },
+  {
+    coinType: 118,
+    decoder: (data: string) => {
```

Comment:
Would you mind refactoring these into generic decode/encode functions akin to `makeChecksummedHexEncoder` and `makeChecksummedHexDecoder`?